### PR TITLE
KAA-1375 [C++ SDK] Failed to parse or compile schema

### PR DESCRIPTION
**Pull request by vadimol**
1) Added default schemes to avoid avrogen errors
2) Updated content of the KaaDefaults.cpp file
3) Added auto-generated files to gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -60,4 +60,11 @@ xcuserdata
...
-doc/Programming-guide/Server-REST-APIs/swagger.json
\ No newline at end of file
+doc/Programming-guide/Server-REST-APIs/swagger.json
+
+# C++ SDK autogenerated files
+client/client-multi/client-cpp/kaa/profile/gen/ProfileGen.hpp
+client/client-multi/client-cpp/kaa/notification/gen/NotificationGen.hpp
+client/client-multi/client-cpp/kaa/log/gen/LogGen.hpp
+client/client-multi/client-cpp/kaa/gen/EndpointGen.hpp
+client/client-multi/client-cpp/kaa/configuration/gen/ConfigurationGen.hpp
```
Comment (rasendubi) [inline .gitignore]: You can put this in `client/client-multi/client-cpp/.gitignore`

**Review by rasendubi — APPROVED** [review]
LGTM

File: .gitignore (hunk 1 above)
Comment (vadimol) [reply .gitignore]: It is a good point. I have updated the PR. Thanks

**Pull request merged**